src/app/map/leaflet-map/leaflet-map.component.ts

```diff
@@ -0,0 +1,239 @@
+              this.markerList.push(marker); // save to list
+              this.markerClusterGroup.addLayer(marker); // save to marker clusters group
+
+              // did we navigate to the map with a poject defined?
+              // if so, zoom to the icon and open the popup
+              if (routerProjId && routerProjId === proj._id) {
+                this.selectedProject = proj;
+                this.map.setView(new L.LatLng(proj.latitude, proj.longitude), 10);
+
+                setTimeout(() => {
+                  this.createMarkerPopup(proj, marker);
+                }, 1000);
+              }
+            }
+          });
+        }
+      });
+    } else {
+      if (this.project.latitude && this.project.longitude) {
+        const title = `Project: ${this.project.name}`;
+        const marker = L.marker(L.latLng(this.project.latitude, this.project.longitude), { title: title }).setIcon(markerIconYellow);
+        marker.projectId = 0;
+        this.markerList.push(marker); // save to list
+        this.markerClusterGroup.addLayer(marker); // save to marker clusters group
+      }
+    }
+  }
+
+  onMarkerClick(...args: any[]) {
+    const proj = args[0] as Project;
+    const marker = args[1].target as L.Marker;
+
+    this.createMarkerPopup(proj, marker);
+  }
+
+  createMarkerPopup(proj: Project, marker: L.Marker) {
+    // this.applist.toggleCurrentApp(app); // update selected item in app list
+
+    // if there's already a popup, delete it
+    let popup = marker.getPopup();
+    if (popup) {
+      const wasOpen = popup.isOpen();
+      popup.remove();
+      marker.unbindPopup();
+      if (wasOpen) { return; }
+    }
+
+    let popupOptions = {};
+    // Fix for different viewports on scrolling for map display
+    if (this.map.getSize().y < 800) {
+      popupOptions = {
+        className: 'map-popup-content',
+        autoPanPaddingTopLeft: L.point(2, 100),
+        autoPanPaddingBottomRight: L.point(2, 30)
+      };
+    } else {
+      popupOptions = {
+        className: 'map-popup-content',
+        autoPanPaddingTopLeft: L.point(80, 200),
+        autoPanPaddingBottomRight: L.point(80, 30)
+      };
+    }
+
+    // compile marker popup component
+    const compFactory = this.resolver.resolveComponentFactory(ProjectPopupComponent);
+    const compRef = compFactory.create(this.injector);
+    compRef.instance.project = proj;
+    compRef.instance.parentMap = this.map;
+    this.appRef.attachView(compRef.hostView);
+    compRef.onDestroy(() => this.appRef.detachView(compRef.hostView));
+    const div = document.createElement('div').appendChild(compRef.location.nativeElement);
+
+    popup = L.popup(popupOptions)
+      .setLatLng(marker.getLatLng())
+      .setContent(div);
+
+    // bind popup to marker so it automatically closes when marker is removed
+    marker.bindPopup(popup).openPopup();
+  }
+
+  @HostListener('unloaded')
+  ngOnDestroy(): void {
+    this.map.remove();
+  }
+
+  ngOnChanges(): void {
```

Comment:
I assume this is needed for an interface? Might be worth commenting why blank.

Reply:
Ah, good catch. I removed the onChange event in place but left the interface in in case I needed it for refreshing. I didn't need it back in the end but did not remove the interface. It's been removed,